sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/TypedAggregateExpression.scala

```diff
@@ -34,6 +34,11 @@ object TypedAggregateExpression {
     val bufferEncoder = encoderFor[BUF]
     val bufferSerializer = bufferEncoder.namedExpressions
     val bufferDeserializer = bufferEncoder.deserializer.transform {
+      case UnresolvedAttribute(nameParts) =>
```

Comment:
good catch! can you try if we can fix it in this way? https://github.com/apache/spark/pull/13269/files#diff-585631b931d2b881e65c585dec11cadbR36

I think it's less hacky